eva/optimizer/optimizer_tasks.py

```diff
@@ -91,6 +91,8 @@ def execute(self):
                     self.optimizer_context.task_stack.push(
                         TopDownRewrite(new_expr, self.rule_set, self.optimizer_context)
                     )
+                    # The root has changed so we cannot apply more rules to the same root, hence return
+                    return
```

Comment:
1. The return is not at the right indention. I think we need to go through all the plan in after.
2. And in a cost-based optimization, multiple rules can be matched to the same root. I don't think we can just return and ignore the rest.

Reply:
1. Rewrite plans should only return 1 output, else it is ambiguous.  I'll add an assert message to take care of it.
2. Rewrites are not part of cost-based optimizations so we are fine. They are taken care of by the state machine.

Reply:
Ah, you are right. Topdown is only rewrite rule not for logical rules. I missed that. Thanks!